### ISSUE TITLE
`TextTester` class based in `MetricTester` class to robustly test `text` metrics

### ISSUE THREAD

**Issue by karthikrangasai** [PyTorchLightning/metrics#411]
## 🚀 Feature

The current `text` package uses various functions to test the `text` metrics. The `MetricTester` class provides a much more robust way to test metrics on all fronts such as multi-device setup etc.

Hence, we can create a new class called `TextTester` based on `MerticTester` that provides us with the necessary tools to robustly test `text` metrics. 

### Motivation

The addition of a new class is required as the inputs to text based metrics are different where they take an `str` or a `Sequence[str]` and not `Tensor` type unlike metrics from other domains.

### Pitch

An implementation of the following functions and class methods that support inputs as text (`str` or `Sequence[str]`) which can be used to test any `text` metric. 
```
def _class_test();

def _functional_test():


class TextTester:
    def run_functional_metric_test():

    def run_class_metric_test():

    @staticmethod
    def run_precision_test_cpu():

    @staticmethod
    def run_precision_test_gpu():

    @staticmethod
    def run_differentiability_test():
```
This will also follow the conversion of the currently existing metrics to `TextTester` based testing.

### Alternatives

We could just use the current function based approach.

**Comment by Borda** [comment]
@karthikrangasai sounds nice! do you want to send a PR? :rabbit:

**Comment by karthikrangasai** [comment]
@Borda yeah, I'm working in it. Got caught up with some college work last week. I will submit one by this week.
